scripts/dts/gen_defines.py

```diff
@@ -359,6 +353,15 @@ def write_compatibles(node):
             f"{node.z_path_id}_COMPAT_MATCHES_{str2ident(compat)}", 1)
 
 
+def write_status(node):
+    status = node.props.get('status')
+
+    if status is None:
+        return
```

Comment:
We can either remove this check or make it a warning as `node.status` should never be none.